guides/common/modules/proc_configuring-repositories.adoc

```diff
@@ -42,22 +42,68 @@ ifdef::katello[]
 ----
 # {package-manager} config-manager --set-enabled powertools
 ----
+
++
+. Enable the Katello module:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# {package-manager} module enable katello:el8
+----
 endif::[]
-ifdef::foreman-el,katello[]
+
+=== In case module enablement fails
+
+If the module fails to enable, it can mean an incorrect module is enabled. List the enabled modules to find out which ones:
```

Comment:
I think my message didn't really come across that well so I'll give the background.

When you run `dnf module enable foreman:el8` (or katello, doesn't really matter) it can fail, but the message that dnf shows can be misleading. It shows warnings about postgresql, but the real reason can be that the Ruby 2.5 module is enabled. Thus users must run `dnf module list --enabled`. If there you see `ruby:2.5` (or really, any ruby module), you must follow the Ruby step. Same if you see the postgresql module. It can also be that both ruby and postgresql are enabled. In that case both steps must be followed.

That is what I'm trying to tell users in a short way. It really isn't that you're attempting to enable an incorrect module, but a module that was previously enabled that blocks enabling of the foreman/katello:el8 module.

Reply:
One thing to consider (but I am not sure if that belongs in this paragraph or not), even if there are no problems, `dnf` produces a warning:

```console
# dnf module enable foreman:el8
Problems in request:
Modular dependency problems with Defaults:

 Problem: module foreman:el8:999920220314215004:6b5f9578.x86_64 requires module(postgresql:12), but none of the providers can be installed
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8040020201220000948:9f9e2e7e.x86_64
  - module postgresql:12:8040020201220000948:9f9e2e7e.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8050020210609162852:b4937e53.x86_64
  - module postgresql:12:8050020210609162852:b4937e53.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8050020210624150605:b4937e53.x86_64
  - module postgresql:12:8050020210624150605:b4937e53.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8050020210716150548:b4937e53.x86_64
  - module postgresql:12:8050020210716150548:b4937e53.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8060020211110235049:d63f516d.x86_64
  - module postgresql:12:8060020211110235049:d63f516d.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8060020211213144215:d63f516d.x86_64
  - module postgresql:12:8060020211213144215:d63f516d.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - conflicting requests
Dependencies resolved.
============================================================================
 Package          Architecture    Version            Repository        Size
============================================================================
Enabling module streams:
 foreman                          el8                                      
 postgresql                       12                                       
 ruby                             2.7                                      

Transaction Summary
============================================================================

Is this ok [y/N]: 
```

The resolution `dnf` comes up with is perfectly fine, but it contradicts the default modules and thus triggers the warning. How do we tell the users that *this* is OK, but others are not?

Especially as the output is almost identical:

```console
# dnf module enable foreman:el8
Last metadata expiration check: 0:03:49 ago on Wed Mar 16 07:34:09 2022.
Error: Problems in request:
Modular dependency problems:

 Problem: module foreman:el8:999920220314215004:6b5f9578.x86_64 requires module(postgresql:12), but none of the providers can be installed
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8040020201220000948:9f9e2e7e.x86_64
  - module postgresql:12:8040020201220000948:9f9e2e7e.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8050020210609162852:b4937e53.x86_64
  - module postgresql:12:8050020210609162852:b4937e53.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8050020210624150605:b4937e53.x86_64
  - module postgresql:12:8050020210624150605:b4937e53.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8050020210716150548:b4937e53.x86_64
  - module postgresql:12:8050020210716150548:b4937e53.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8060020211110235049:d63f516d.x86_64
  - module postgresql:12:8060020211110235049:d63f516d.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - module postgresql:10:8040020201214234117:9f9e2e7e.x86_64 conflicts with module(postgresql:12) provided by postgresql:12:8060020211213144215:d63f516d.x86_64
  - module postgresql:12:8060020211213144215:d63f516d.x86_64 conflicts with module(postgresql:10) provided by postgresql:10:8040020201214234117:9f9e2e7e.x86_64
  - conflicting requests
```

The only differences are in `Modular dependency problems with Defaults:` vs `Modular dependency problems:` in the initial message and the fact that there is no suggested resolution.

Reply:
@ekohl I'm sorry, I misread that. In that case, I have a different solution. Hang on.

Reply:
Thanks @evgeni for sharing the explicit output. You can indeed see how similar they are and it's **very** easy for a user to misread the output (I know I had issues). It was my intent here to address that.